### PR TITLE
fix(CLI): rename update to update-deprecations

### LINKED ISSUE #18: Change the name `packwerk update` to `packwerk update-deprecations`
@exterm brought up that `packwerk update` should actually be called `packwerk update-deprecations` and I totally agree. `packwerk update` can be confusing. 

We should add a `packwerk update-deprecations` command and add a deprecation warning for the `packwerk update` command but still allow people to use it.

### PR DESCRIPTION
**Pull request by hendrixfan**
- fixes #18
- wrapped the update method because it still needs to work
- added deprecations warning and tested output

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code refactor (non-breaking change that doesn't add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: lib/packwerk/cli.rb [hunk 1]
```diff
@@ -52,7 +54,8 @@ def execute_command(args)
           Subcommands:
             init - set up packwerk
             check - run all checks
-            update - update deprecated references
+            update - update deprecated references (deprecated use update-deprecations instead)
```
Comment (exterm) [inline lib/packwerk/cli.rb:57]:
```suggestion
            update - update deprecated references (deprecated, use update-deprecations instead)
```

File: lib/packwerk/cli.rb [hunk 2]
```diff
@@ -117,6 +120,11 @@ def generate_configs
     end
 
     def update(paths)
+      warn("`packwerk update` will be deprecated in favor of `packwerk update-deprecations`.")
```
Comment (exterm) [inline lib/packwerk/cli.rb:123]:
```suggestion
      warn("`packwerk update` is deprecated in favor of `packwerk update-deprecations`.")
```

AFAIK `deprecated` is already forward-looking; it means that it's going to be removed in the future

**Comment by exterm** [comment]
Hey Wolfgang, thank you for your contribution!

I added a few comments, and I think `USAGE.md` also needs to be updated.

**Comment by exterm** [comment]
Same with `TROUBLESHOOT.md`.

**Comment by hendrixfan** [comment]
hej @exterm, I've fixed it and squashed the commits

**Review by exterm — APPROVED** [review]

**Comment by exterm** [comment]
Great work! We'll need you to [sign the CLA](https://cla.shopify.com/) before we can merge this though.

**Comment by hendrixfan** [comment]
@exterm I did that already. I think the CLA Check needs to restart.

**Comment by exterm** [comment]
Oh no! We're having strange problems right now with all kinds of statuses not updating. Hold on. We'll fix.

**Comment by wildmaples** [comment]
Hi @hendrixfan ! Would you mind trying to rebase and push to see if CI gets triggered? Thanks!

**Comment by hendrixfan** [comment]
hej @wildmaples. Done

**Comment by wildmaples** [comment]
Just noticed this PR did not follow the PR template and contains a deprecation. Users should be switching over to use `packwerk update-deprecations` in the next release.

**Pull request merged**